aee.js

```diff
@@ -23,3 +23,15 @@ Array.prototype.any = function (spec) {
   }
   return false
 }
+
+Array.prototype.take = function (howMany, spec) {
+  let result = []
+  if (howMany <= 0) return result
+  const len = this.length
+  const searchSpec = (typeof spec === 'function') ? spec : () => true
```

Comment:
Yeah, today actually, I learned that `splice()` is a destructive function, and it messed up my tests by modifying the original data, even though it was declared as `const`. Also learned a bit about using hooks in mocha/chai like `before()`, `afterEach()` and how to use them. But instead chose to use `slice()`, and only keep one reference to `data` in the whole test file. What do you think of this approach?